core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/IssueStorePaths.java

```diff
@@ -33,61 +36,35 @@
 
 public class IssueStorePaths {
 
-  public String sqPathToFileKey(Sonarlint.ProjectConfiguration projectConfiguration, String projectKey, String sqFilePath) {
-    var modulePaths = projectConfiguration.getModulePathByKeyMap();
-
-    // find longest prefix match
-    var subModuleKey = projectKey;
-    var prefixLen = 0;
-
-    for (Map.Entry<String, String> entry : modulePaths.entrySet()) {
-      var entryModuleKey = entry.getKey();
-      var entryPath = entry.getValue();
-      if (!entryPath.isEmpty() && sqFilePath.startsWith(entryPath) && prefixLen <= entryPath.length()) {
-        subModuleKey = entryModuleKey;
-        prefixLen = entryPath.length() + 1;
-      }
-    }
-
-    var relativeFilePath = sqFilePath.substring(prefixLen);
-    return subModuleKey + ":" + relativeFilePath;
-  }
-
   @CheckForNull
-  public String idePathToFileKey(Sonarlint.ProjectConfiguration projectConfiguration, ProjectBinding projectBinding, String ideFilePath) {
+  public String idePathToFileKey(ProjectBinding projectBinding, String ideFilePath) {
     var sqFilePath = idePathToSqPath(projectBinding, ideFilePath);
 
     if (sqFilePath == null) {
       return null;
     }
-    return sqPathToFileKey(projectConfiguration, projectBinding.projectKey(), sqFilePath);
+    return projectBinding.projectKey() + ":" + sqFilePath;
   }
 
   @CheckForNull
-  public String idePathToSqPath(ProjectBinding projectBinding, String ideFilePath) {
-    if (!ideFilePath.startsWith(projectBinding.idePathPrefix())) {
-      return null;
-    }
-    var localPrefixLen = projectBinding.idePathPrefix().length();
-    if (localPrefixLen > 0) {
-      localPrefixLen++;
-    }
-    var sqPathPrefix = projectBinding.sqPathPrefix();
-    if (!sqPathPrefix.isEmpty()) {
-      sqPathPrefix = sqPathPrefix + "/";
+  public String idePathToSqPath(ProjectBinding projectBinding, String ideFilePathStr) {
+    var ideFilePath = Paths.get(ideFilePathStr);
```

Comment:
As you changed the implementation to use `Paths.get()`, I wondered if we could get an exception at some point. But I think all the provided parameters are valid relative paths.

Reply:
We are already using `Path` in the prefix computation algorithm, so I think it is pretty safe.